app/bundles/FormBundle/Model/SubmissionModel.php

```diff
@@ -326,18 +326,19 @@ public function saveSubmission($post, $server, Form $form, Request $request = nu
         if (!empty($leadFieldMatches)) {
             $lead = $this->createLeadFromSubmit($form, $leadFieldMatches, $leadFields);
             $submission->setLead($lead);
-        }
-
-        // Get updated lead if applicable with tracking ID
-        if ($form->isInKioskMode()) {
-            $lead = $this->leadModel->getCurrentLead();
         } else {
-            list($lead, $trackingId, $generated) = $this->leadModel->getCurrentLead(true);
+            // Get updated lead if applicable with tracking ID
```

Comment:
Could you share reasoning behind this change? It'll cause the tracking ID to not be associated with the submission which breaks A/B tests.